opencensus/proto/trace/v1/trace.proto

```diff
@@ -47,22 +47,31 @@ message Span {
   // This field is required.
   bytes span_id = 2;
 
-  // The `tracestate` field conveys information about request position in multiple distributed
-  // tracing graphs.
-  //
-  // There can be a maximum of 32 members in the map.
-  //
-  // The key must begin with a lowercase letter, and can only contain lowercase letters 'a'-'z',
-  // digits '0'-'9', underscores '_', dashes '-', asterisks '*', and forward slashes '/'. For
-  // multi-tenant vendors scenarios '@' sign can be used to prefix vendor name. The maximum length
-  // for the key is 256 characters.
-  //
-  // The value is opaque string up to 256 characters printable ASCII RFC0020 characters (i.e., the
-  // range 0x20 to 0x7E) except ',' and '='. Note that this also excludes tabs, newlines, carriage
-  // returns, etc.
+  // This field conveys information about request position in multiple distributed tracing graphs.
+  // It is a list of Tracestate.Entry with a maximum of 32 members in the list.
   //
   // See the https://github.com/w3c/distributed-tracing for more details about this field.
-  map<string, string> tracestate = 15;
+  message Tracestate {
```

Comment:
Minor nit, why don't we break this out into a message of its own, that is, a top-level definition instead of an inlined definition here?